spec/models/account_spec.rb

```diff
@@ -59,72 +67,147 @@
+    before do
+      stub_request(:post, "https://some_url.com/oauth/token").with(
+        body: {
+          "client_id" => "some_client_id",
+          "client_secret" => "some_client_secret",
+          "grant_type"=>"client_credentials"
+        },
+        headers: {
+          "Accept" => "application/vnd.api+json",
+          "Content-Type" => "application/x-www-form-urlencoded"
+        }
+      ).to_return(
+        status: 200,
+        body: { "access_token": "the_access_token" }.to_json,
+        headers: { "Content-Type" => "application/json" }
+      )
+    end
 
-      account.clear_token!
-      account.reload
+    it "returns a client credential token setup without default params" do
+      expect(account.application_token.token).to eq "the_access_token"
+    end
+  end
+
+  describe "#oauth_client" do
+    let!(:account) { Account.create!(synced_id: 123) }
 
-      expect(account.oauth_access_token).to be_nil
-      expect(account.oauth_refresh_token).to be_nil
-      expect(account.oauth_expires_at).to be_nil
+    it "returns a BookingSync::Engine.oauth_client setup without default params" do
+      expect(account.oauth_client).to be_an OAuth2::Client
+      expect(account.oauth_client.id).to eq "some_client_id"
+      expect(account.oauth_client.secret).to eq "some_client_secret"
+    end
+  end
+
+  describe "#application" do
+    let!(:account) { Account.create!(synced_id: 123) }
+
+    it "returns nil" do
+      expect(account.application).to be_nil
     end
   end
 
   describe "#api" do
-    it "returns API client initialized with OAuth token" do
-      token = double(token: "access_token", expired?: false)
-      allow(OAuth2::AccessToken).to receive(:new)
-        .and_return(token)
-      account = Account.new
+    let!(:account) { Account.new(oauth_access_token: "access_token") }
 
+    it "returns API client initialized with OAuth token" do
       expect(account.api).to be_kind_of(BookingSync::API::Client)
       expect(account.api.token).to eq("access_token")
     end
   end
+
+  describe "#clear_token!" do
+    it "clears token related fields on account" do
+      account = Account.create!(oauth_access_token: "token",
+        oauth_refresh_token: "refresh", oauth_expires_at: "expires")
+
+      expect { account.clear_token! }
+        .to change { account.reload.oauth_access_token }.from("token").to(nil)
+        .and change { account.oauth_refresh_token }.from("refresh").to(nil)
+        .and change { account.oauth_expires_at }.from("expires").to(nil)
+    end
+  end
+
+  describe "#update_token" do
+    it "updates the token related fields on account" do
+      token = double(token: "new_access_token", refresh_token: "new_refresh_token", expires_at: "new_expires_at")
+      account = MultiApplicationsAccount.create!(oauth_access_token: "token",
+        oauth_refresh_token: "refresh", oauth_expires_at: "expires", host: "example.test")
+
+      expect { account.update_token(token) }
```

Comment:
also, would be nice to move these statements to `subject`s

Reply:
For 1 test case maybe better to keep it like this for readability.